controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +74,39 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isEnableTls()) {
+                SSLContextConfigurator sslCon = new SSLContextConfigurator();
+                sslCon.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                sslCon.setKeyStorePass(loadPasswdFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(sslCon, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
+        } catch (Exception e) {
+          log.error("Exception while creating HTTP server", e);
```

Comment:
This exception needs to be rethrown. Otherwise the `AbstractIdleService` will have started up normally but there's no server to back it up.

Reply:
Fixed. Dropping the `catch` block so that the earlier behavior is maintained.